client/client.go

```diff
@@ -66,6 +70,77 @@ func NewClient(baseUrl, authHeader string) *Client {
 	}
 }
 
+type APIError struct {
+	StatusCode int
+	Message    string
+}
+
+func (e *APIError) Error() string {
+	return fmt.Sprintf("API error (status %d): %s", e.StatusCode, e.Message)
+}
+
+func handleAPIError(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	if apiErr, ok := err.(*runtime.APIError); ok {
+        return &APIError{
+            StatusCode: apiErr.Code,
+            Message:    apiErr.Error(),
+        }
+    }
+
+	errStr := err.Error()
```

Comment:
From here to the end of this function can also be rewritten with `errors.As`. All of the generated error types implement this interface:

```go
type GeneratedAPIError interface {
	Code() int
	Message() string
}
```

You can use `errors.As` to with that (or a similar) interface type, and then you can just call the `Code()` and `Message()` as normal.

Reflection is used internally by the Go standard library quite often, but you'll rarely need to use it yourself. There's usually a more idiomatic solution.